docs/README.md

```diff
@@ -1,3 +1,201 @@
+| `Swift.UInt16`                  | `ushort` |
+| `Swift.Int8`                    | `sbyte`  |
+| `Swift.UInt8`                   | `byte`   |
+| `Swift.UnsafeRawPointer`        | `IntPtr` |
+| `Swift.UnsafeMutableRawPointer` | `IntPtr` |
+| `Int`                           | `nint`   |
+| `UInt`                          | `nuint`  |
+| `Bool`                          | `bool`   |
+| `Float`                         | `float`  |
+| `Double`                        | `double` |
+
+Swift primitive types are implemented as frozen structs that conform to Swift-specific lowering processes handled by the runtime. However, such mappingcan fit within the underlying calling convention as these types are below the size limit for being passed by reference. 
+
+### Structs
+
+Swift structs are projected into C# as `IDisposable` classes, implementing `ISwiftStruct`, which inherit from `ISwiftNominalType` and `ISwiftDisposable`. This approach is chosen due to the semantic differences in lifecycle management between Swift and C# types. Bindings are generated with `SwiftStructAttribute` that inherits `SwiftNominalTypeAttribute`. This attribute contains information about the Swift type.
+
+Given the following Swift struct declaration:
+```swift
+    public struct BarInt {
+        public var X:Int; 
+        public init(x:Int) {
+            X = x;
+        }
+    }
+```
+
+The projection tooling will generate the following C#, with function bodies left empty for simplicity:
+```csharp
+    using System;
+    using System.Runtime.InteropServices;
+    using SwiftRuntimeLibrary;
+    using SwiftRuntimeLibrary.SwiftMarshal;
+    
+    namespace NewClassCompilerTests
+    {
+        [SwiftStruct("libNewClassCompilerTests.dylib",
+            "$s21NewClassCompilerTests6BarIntVMn", 
+            "$s21NewClassCompilerTests6BarIntVN", "")]
+        public class BarInt : ISwiftStruct
+        {
+            public BarInt(nint x)
+            {
+            }
+            internal BarInt(SwiftNominalCtorArgument unused)
+            {
+            }
+            public static SwiftMetatype GetSwiftMetatype()
+            {
+            }
+            public void Dispose()
+            {
+            }
+            void Dispose(bool disposing)
+            {
+            }
+            ~BarInt()
+            {
+            }
+            public byte[] SwiftData
+            {
+                get; set;
+            }
+            public nint X
+            {
+                get { }
+                set { }
+            }
+        }
+    }
+```
+
+There is a payload `SwiftData` along with two constructors. The first maps onto the `init` method inside the swift class. The second is an internal constructor that gets used to define an uninitialized type. This constructor gets used by the marshaler when a type needs to be allocated before it gets used, for example, as a return value because Swift semantics don’t allow to explicitly have variables in an uninitialized state. 
```

Comment:
> This constructor gets used by the marshaler when a type needs to be allocated before it gets used, for example, as a return value because Swift semantics don’t allow to explicitly have variables in an uninitialized state.

We aren't going to be able to pass `class` types down to Swift so I don't see why this is needed, perhaps it makes the source generation easier? For example, I would consider the two following scenarios - either the type has an official default constructor or doesn't and we create an `internal` default constructor.

What does the constructor with a `SwiftNominalCtorArgument` get us?

Reply:
> We aren't going to be able to pass class types down to Swift so I don't see why this is needed, perhaps it makes the source generation easier?

How are we supposed to interop with functions that accept `class` types?

> What does the constructor with a `SwiftNominalCtorArgument` get us?

According to the docs, this constructor should be used by marshaling; it allocates the `SwiftData` member to the correct size but does not initialize it.

Reply:
> How are we supposed to interop with functions that accept class types?

They will be reduced to a raw pointer at some point. There is no way in CoreCLR to marshal a `class` instance across an interop boundary without some kind of projection. I agree that the user facing API surface will accept a `class` instance, but the generated code will convert that into a blittable form - likely a raw pointer.

> According to the docs, this constructor should be used by marshaling; it allocates the SwiftData member to the correct size but does not initialize it.

If we need to allocate a managed object projection in an uninitialized state the marshaller should use something akin to [`RuntimeHelpers.GetUninitializedObject()`](https://learn.microsoft.com/dotnet/api/system.runtime.compilerservices.runtimehelpers.getuninitializedobject).

I could be missing nuance here on how this all comes together so if this doesn't align with the design let's discuss it with some concrete examples we can walk though.